### PR TITLE
use Azure::Core::ETag for ETag type

### PR DESCRIPTION
**Pull request by Jinming-Hu**
closes https://github.com/Azure/azure-sdk-for-cpp/issues/1484

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Comment by Jinming-Hu** [comment]
@katmsft Please enbale DataLake and file serivce

**Review by katmsft — APPROVED** [review]

File: sdk/storage/azure-storage-files-datalake/CHANGELOG.md [hunk 1]
```diff
@@ -40,6 +40,8 @@
 - Changed `SetAccessControlRecursive` to `SetAccessControlRecursiveListSinglePage`, to mark that it is a single page operation, and removed the `mode` parameter, separated the modify/delete functionality to two new APIs.
 - Moved `SetAccessControlRecursiveListSinglePage` to `DataLakePathClient`.
 - Changed `MaxRecord` to `MaxEntries`, `ForceFlag` to `ContinueOnFailure` to be more accurate names.
+- Type for ETag was changed to `Azure::Core::ETag`.
+- Type for `IfMatch` and `IfNoneMatch` was changed to `Azure::Core::ETag`.
```
[the remark this answers is not in the document]
Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/CHANGELOG.md:44]: I think `ETag` was just the name of the member variable before this change. It is different than String or Array or any other well-known type.

[the remark this answers is not in the document]
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-datalake/CHANGELOG.md:44]: whatever

**Pull request merged**